.github/workflows/codegen-ci.yml

```
@@ -0,0 +1,146 @@
+name: Typescript SDK CI
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      # first three are default, adding labeled for "CI:TEST" support
+      # (see jobs.unit.if below)
+      - labeled
+    paths:
+      - packages/sdk-codegen/**
+      - packages/sdk-codegen-utils/**
+      - packages/sdk-codegen-scripts/**
+
+  push:
+    branches:
+      - main
+    paths:
+      - packages/sdk-codegen/**
+      - packages/sdk-codegen-utils/**
+      - packages/sdk-codegen-scripts/**
+
+env:
+  LOOKERSDK_BASE_URL: https://localhost:20000
+  LOOKERSDK_VERIFY_SSL: false
+  TS_JUNIT_OUTPUT_DIR: results/sdk-codegen
+
+jobs:
+  unit:
+    # 1st condition: push event (restricted to main above)
+    # 2nd condition: PR opened/synchronized/reopened (types filter above)
+    #                unless the PR has the CI:HOLD label
+    # 3rd condition: PR labeled with "CI:TEST"
+    if: >
+      github.event_name == 'push' ||
+      (
+        github.event_name == 'pull_request' &&
+        github.event.action != 'labeled' &&
+        !contains(github.event.pull_request.labels, 'CI:HOLD')
+      ) ||
+      (
+        github.event_name == 'pull_request' &&
+        github.event.action == 'labeled' &&
+        github.event.label.name == 'CI:TEST'
+      )
+
+    name: Unit - ${{ matrix.os }} / Node ${{ matrix.node-version }}
+    env:
+      JEST_JUNIT_OUTPUT_DIR: results/sdk-codegen
+      JEST_JUNIT_OUTPUT_NAME: ${{ matrix.os }}.sdk-codegen${{ matrix.node-version }}.xml
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        os:
+          - ubuntu
+        node-version: [12.x, 14.x, 15.x]
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Repo Checkout
+        uses: actions/checkout@v2
+
+      - name: Install Node ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install dependencies
+        run: |
+          yarn
+          yarn lint:es --quiet
+          yarn build
```

```suggestion
          yarn build
          yarn lint:es --quiet
```

build should be attempted first to ensure import references for linting are successful.